Tests/Application/Testing/HeadlessBundleKernelBrowser.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\HeadlessBundle\Tests\Application\Testing;
+
+use Symfony\Bundle\FrameworkBundle\KernelBrowser;
+use Symfony\Component\DomCrawler\Crawler;
+
+/**
+ * @deprecated This class has only be introduced to keep BC between Symfony 5.2 and previous versions.
+ *
+ * Will be removed, as soon as the min-requirement of sulu/sulu is high enough for the `SuluKernelBrowser` to be always available.
+ *
+ * @internal
+ */
+class HeadlessBundleKernelBrowser extends KernelBrowser
```

Comment:
This should not be used. The headless bundle works differently to other API's it need to ignore the `Accept` header and always return the `.json` also when the browser does not send the `Accept` header with introducing this KernelBrowser we are not longer testing the HeadlessBundle correctly. This KernelBrowser should be removed and `jsonRequest` should not be used to test the HeadlessBundle endpoints. Detection on the format is just done by the `.json` at the end.

Reply:
But exceptions are returned as symfony exception page and not as json ... and an url like `/api/snippet-areas/default.json` won't work, because then symfony assumes `default.json` is the `{area}` instead of `default`

Reply:
So how can i test the exceptions then?

Reply:
The exception should still be json. Seems then the `_format` is not correctly set for this route try add something like:

```yaml
     defaults:
          _format: json
```

Check if in the controller `$request->getRequestFormat()` returns you `json`. For navigation and snippet its okay to have `.json` optional e.g.: `/api/snippet-areas/{area}.{_format}` but the  `$request->getRequestFormat()` should return json.

Reply:
Did add the same also to the navigation route